internal/plugins/ansible/v1/scaffolds/api.go

```diff
@@ -37,53 +34,54 @@ import (
 	ansibleroles "github.com/operator-framework/operator-sdk/internal/plugins/ansible/v1/scaffolds/internal/templates/roles"
 )
 
-var _ cmdutil.Scaffolder = &apiScaffolder{}
+var _ plugins.Scaffolder = &apiScaffolder{}
 
 type apiScaffolder struct {
+	fs afero.Fs
+
 	config   config.Config
-	resource *resource.Resource
+	resource resource.Resource
 
 	doRole, doPlaybook bool
 }
 
-// NewCreateAPIScaffolder returns a new Scaffolder for project initialization operations
-func NewCreateAPIScaffolder(config config.Config, res *resource.Resource, doRole, doPlaybook bool) cmdutil.Scaffolder {
+// NewCreateAPIScaffolder returns a new plugins.Scaffolder for project initialization operations
+func NewCreateAPIScaffolder(cfg config.Config, res resource.Resource, doRole, doPlaybook bool) plugins.Scaffolder {
 	return &apiScaffolder{
-		config:     config,
+		config:     cfg,
 		resource:   res,
 		doRole:     doRole,
 		doPlaybook: doPlaybook,
 	}
 }
 
-func (s *apiScaffolder) newUniverse(r *resource.Resource) *model.Universe {
-	return model.NewUniverse(
-		model.WithConfig(s.config),
-		model.WithResource(r),
-	)
+// InjectFS implements plugins.Scaffolder
+func (s *apiScaffolder) InjectFS(fs afero.Fs) {
+	s.fs = fs
 }
 
-// Scaffold implements Scaffolder
+// Scaffold implements plugins.Scaffolder
 func (s *apiScaffolder) Scaffold() error {
-	return s.scaffold()
-}
-
-func (s *apiScaffolder) scaffold() error {
-	if s.resource == nil {
-		return errors.New("resource must not be nil")
-	}
-
-	if err := s.config.UpdateResource(*s.resource); err != nil {
+	if err := s.config.UpdateResource(s.resource); err != nil {
 		return err
 	}
 
-	var createAPITemplates []file.Builder
+	// Initialize the machinery.Scaffold that will write the files to disk
+	scaffold := machinery.NewScaffold(s.fs,
+		// NOTE: kubebuilder's default permissions are only for root users
```

Comment:
nit: we might cann complement with about the error face.. to describe better why ansible requires that.

Reply:
Will try to find the yielded error and add it